### DISCUSSION TITLE
is there a way to get a url instead of making a request with .query?

### DISCUSSION THREAD

**Discussion by vanillacode314** [trpc/trpc#3013] [Q&A]
I want to build a url instead of querying the server as I want to pass the URL to an image tag. Is this possible?

**Comment by JoelBrenstrum** [comment]
I would also like this.  We also expose image data over an express endpoint and render the URL directly into the `src` attribute.

We also have Playwright API tests in our project and it would be nice to consume the client to generate the correct URL and inputs rather than typing them manually.

**Reply by KATT** [reply]
How would you pass the URL to an image tag? We only support JSON as the response

**Reply by JoelBrenstrum** [reply]
Right, that means it wouldn't currently be suitable for us. We have a few endpoints that return Blobs (PDF, Image).
It might be nice in the future to have a way to return a raw response.

Getting the plain URL would still be nice for things like playwright tests, or even when calling out to another service when wanting to pass a URL to hit. We use Google Cloud Tasks which require the URL you wish to have the task call off to.

I haven't dug too deep, but a `type` for the response data would be nice since the data is unwrapped in the tRPC layer.
